library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -132,11 +131,17 @@ private void updateNotification(DownloadBatchStatus liteDownloadBatchStatus, Dow
                 (int) liteDownloadBatchStatus.bytesDownloaded()
         );
 
-        downloadService.updateNotification(notificationInformation);
+        if (liteDownloadBatchStatus.status() == DELETION) {
+            downloadService.dismissNotification();
+            return;
+        }
 
-        if (liteDownloadBatchStatus.status() != DOWNLOADING) {
-            downloadService.updateNotification(notificationInformation);
+        if (liteDownloadBatchStatus.status() == DOWNLOADED) {
+            downloadService.stackNotification(notificationInformation);
+            return;
         }
+
+        downloadService.updateNotification(notificationInformation);
```

Comment:
Can I get a clarification here. 

We are updating the service with information, but if the status was `DELETION` we would `dismiss()` before we `update()` . Is this the desired behaviour?

Reply:
No that's not what happens on `DELETION`. If we `DELETE` the notification just gets dismissed, there is no update because of the return.

Reply:
Very good point 👍